sdk/core/core-client/src/deserializationPolicy.ts

```diff
@@ -118,24 +123,110 @@ async function deserializeResponseBody(
   }
 
   const responseSpec = getOperationResponseMap(parsedResponse);
-  const expectedStatusCodes = Object.keys(operationSpec.responses);
-  const hasNoExpectedStatusCodes =
-    expectedStatusCodes.length === 0 ||
-    (expectedStatusCodes.length === 1 && expectedStatusCodes[0] === "default");
-  const isExpectedStatusCode: boolean = hasNoExpectedStatusCodes
-    ? 200 <= parsedResponse.status && parsedResponse.status < 300
-    : !!responseSpec;
+  const errorResponse = handleErrorResponse(parsedResponse, operationSpec);
```

Comment:
Can this just be a `void` function which only throws if the response matches any error?

Reply:
Changed the method to return 2 values based on Jeff's suggestion.